packages/client/hmi-client/src/model-representation/service.ts

```diff
@@ -365,3 +365,17 @@ export function cleanModel(model: Model): void {
 		}
 	});
 }
+
+export function isModelMissingMetadata(model: Model): boolean {
+	const parameters: ModelParameter[] = getParameters(model);
+	const initials: Initial[] = getInitials(model);
+
+	const initialsCheck = initials.some((i) => {
+		const initialMetadata = getInitialMetadata(model, i.target);
+		return !initialMetadata?.name || !initialMetadata?.description || !initialMetadata?.units;
+	});
+
+	const parametersCheck = parameters.some((p) => !p.name || !p.description || !p.units?.expression);
+
+	return initialsCheck || parametersCheck;
+}
```

Comment:
checking initials and parameters for the field that are shown on model config

Reply:
fyi you'll probably have to edit this later since @mwdchang told me that for initials I should be getting the `name`, `description` etc from `model.states` instead of storing everything in metadata

Reply:
Actually I can edit this once I do this with states

Reply:
🙏